### PR TITLE
chore: Improve `to_alternative_strings` performance

### PR DESCRIPTION
**Pull request by blt**
This PR is a series of commits to improve the performance of `to_alternative_strings`. I am examining this function as it's a hotspot in `file -> remap -> blackhole` performance tests. Please see individual commits for details, performance bumps. 

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

File: lib/vrl/compiler/benches/path.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+];
+
+impl fmt::Display for Parameters {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.basis)
+    }
+}
+
+fn benchmark_to_alternative_strings(c: &mut Criterion) {
+    let mut group = c.benchmark_group("path::to_alternative_strings");
```
Comment (jszwedko) [inline lib/vrl/compiler/benches/path.rs:29]: Could we namespace this under `vrl`? That makes the aggregate output of the benchmark suites easier to grok.

File: lib/vrl/compiler/benches/path.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    for param in &PARAMETERS {
+        group.throughput(Throughput::Bytes(param.basis.len() as u64));
+
+        group.bench_with_input(BenchmarkId::from_parameter(param), &param, |b, &param| {
+            let path = Path::from_str(param.basis).unwrap();
+            b.iter(|| {
+                let _ = path.to_alternative_strings();
```
Comment (jszwedko) [inline lib/vrl/compiler/benches/path.rs:36]:
```suggestion
                path.to_alternative_strings()
```

I'm unsure if the compiler actually does this, but I could see it optimizing that call away. Returning the value causes criterion to wrap it in `black_box`.

**Review by jszwedko — COMMENTED** [review]
@blt makes sense! Do you have before/after benches?

**Review by JeanMertz — APPROVED** [review]
Thanks for this improvement. This looks good to me 👍

I should note that all `*alternative*` functions here are temporary hacks until #5374 lands.

But still, between now and then, it's nice to have these improvements in.

File: lib/vrl/compiler/benches/path.rs (hunk 1 above)
Comment (blt) [reply lib/vrl/compiler/benches/path.rs:29]: Sure thing.

**Comment by blt** [comment]
@jszwedko good question. The initial commit shows that the runtime compared to before the benchmark was introduced, but I didn't make a clean measurement. The initial commit is +20%/+10%/+2% over baseline. Second benchmark is +10%/+20%/+60% over that. Third is +2/+0/+10, fourth is +1/+10/+7. This change ought to be +30/+30/+70 give or take but in the future I'll be careful to get a baseline before making changes. I can't find one in my shell history, but it ought to have been captured in the commit history.

File: lib/vrl/compiler/benches/path.rs (hunk 2 above)
Comment (blt) [reply lib/vrl/compiler/benches/path.rs]: Incorporated.

**Review by jszwedko — APPROVED** [review]
Awesome!

**Pull request merged**